src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/StructBuilder.java

```diff
@@ -49,104 +54,117 @@
         this.structLayout = structLayout;
         this.structType = structType;
         prefixElementNames = new ArrayDeque<>();
+        classBegin();
```

Comment:
Kinda strange that the constructor here call `classBegin()` but OutputFactory calls `classEnd()`. Could both calls be moved to `OutputFactory`?

Reply:
I can fix this

Reply:
forgot about this - I cannot really do this, as nested structs don't return a new builder - in other words, the OutputFactory code doesn't really know if it should begin a class or not. I could do `if (newBuilder != currentBuilder) newBuilder.classBegin() ...` but I don't think that improves much?

Reply:
Yeah, I was looking at `classEnd()` where you use `if (prefixElementNames.isEmpty())` to check if the struct builder is for an anonymous nested struct or not. So I thought you could do the same for `classBegin()` and have:

```
    @Override
    void classBegin() {
        if (prefixElementNames.isEmpty()) {
            super.classBegin();
            addLayout(layoutField(), ((Type.Declared)structType).tree().layout().get())
                    .emitGetter(this, MEMBER_MODS, Constant.SUFFIX_ONLY);
        }
    }
```

?

Reply:
Also, it might make things clearer to create an `isAnonymousNested()` predicate for that (just for reader's sake).

Reply:
Not sure about the predicate - in the sense that the builder is not an anonymous nested - it is just being reused/recycled for an anonymous nested.